src/dune/dune_file.ml

```diff
@@ -1720,11 +1720,31 @@ end
 module Copy_files = struct
   type t =
     { add_line_directive : bool
+    ; alias : Alias.Name.t option
+    ; mode : Rule.Mode.t
     ; glob : String_with_vars.t
     ; syntax_version : Dune_lang.Syntax.Version.t
     }
 
-  let decode = String_with_vars.decode
+  let long_form =
+    let+ alias = field_o "alias" Alias.Name.decode
+    and+ mode = field "mode" ~default:Rule.Mode.Standard Rule.Mode.decode
+    and+ glob = field "glob" String_with_vars.decode
+    and+ syntax_version = Dune_lang.Syntax.get_exn Stanza.syntax in
+    { add_line_directive = false; alias; mode; glob; syntax_version }
+
+  let decode =
+    peek_exn >>= function
+    | List _ -> Dune_lang.Syntax.since Stanza.syntax (2, 7) >>> fields long_form
```

Comment:
This gives
```
Error: 'copy_files' is only available since version 2.7 of the dune language.
Please update your dune-project file to have (lang dune 2.7).
```
as error message, which is not exactly what we want; suggestions appreciated.

Reply:
How about:

> Error: (alias) and (mode) are available in this position only since version 2.7 of the dune language.
> Please update your dune-project file to have (lang dune 2.7)

You can also have a look at how `(enabled_if)` behaves, we added support for it in more positions later on, so the error message might be similar.

Reply:
Thanks for the hint, I did something similar to what is done in other places.